RELEASE.md

```diff
@@ -0,0 +1,52 @@
+# Release Process for `connect-go`
```

Comment:
Please use title case everywhere for headings (not going to comment on every instance here)
```suggestion
# Release process for `connect-go`
```